lib/motors/controller.cpp

```diff
@@ -2,8 +2,142 @@
+    case 0xFF0B:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_MOTOR_FEEDBACK: Wrong feedback selected (check feedback type)");
+      break;
+    case 0xFF0C:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_DC_LINK_UNDERVOLTAGE: DC voltage not applied to bridge or too low");
+      break;
+    case 0xFF0D:
+      logger_.log(core::LogLevel::kFatal, "ERROR_PULS_MODE_FINISHED: Pulse mode finished");
+      break;
+    case 0xFF0E:
+      logger_.log(core::LogLevel::kFatal, "ERROR_APP_ERROR");
+      break;
+    case 0xFF0F:
+      logger_.log(core::LogLevel::kFatal, "ERROR_EMERGENCY_BUTTON: Emergency button pressed");
+      break;
+    case 0xFF10:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_CONTROLLER_OVERTEMPERATURE: Controller overtemperature");
+      break;
+    case 0x3210:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_DC_LINK_OVERVOLTAGE: Power supply voltage too high");
+      break;
+    default:
+      logger_.log(core::LogLevel::kFatal,
+                  "GENERIC_ERROR: Unspecific error occurred with code %i",
+                  error_code);
+      break;
+  }
+}
+
+ControllerStatus Controller::processWarningMessage(const std::uint8_t warning_code)
+{
+  // Flag specific warnings in binary
+  if (warning_code & 0x2) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: Motor Temperature Exceeded (code: %x)",
+                warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x4) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: DC link under voltage (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x8) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: DC link over voltage (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x10) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: DC over priority_errorent (code: %x)",
+                warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x20) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: Stall protection active (code: %x)",
+                warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x40) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: Max velocity exceeded (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x80) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: BMS Proposed Power (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x100) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: Capacitor temporature exceeded (code: %x)",
```

Comment:
I'm not sure what capacitors this is talking about but possibly they are also something we could increase cooling to deal with

Reply:
Also temperature is spelled wrong

Reply:
Do we want this to be made into another ControllerStatus enum? or shall we just leave as fatal

Reply:
and i just noticed controller is spelled wrong in every warning message log